plugins/search-react/src/components/SearchResultList/SearchResultList.tsx

```diff
@@ -91,6 +113,10 @@ export type SearchResultListProps = Omit<
    * A search query used for requesting the results to be listed.
    */
   query: Partial<SearchQuery>;
+  /**
+   * Optional property to provide if component should not render the group when no results are found.
+   */
+  disableRenderingWithNoResults?: boolean;
```

Comment:
Ahh--I was having trouble figuring out the purpose behind this prop.  But I guess it's only used if you explicitly want nothing to be rendered when there are no results, vs. a "no results" empty state (for which there is a default).

Would it be too unintuitive to get rid of this prop, but allow the `noResultsComponent` prop to be set to `null` in order to trigger this behavior?

So the three possible states are...

``` tsx
// Default empty state on no results
<SearchResultList
  query={query}
/>

// Custom empty state on no results
<SearchResultList
  query={query}
  noResultsComponent={<ListItemText primary="No results were found" />}
/>

// Do not render on no results
<SearchResultList
  query={query}
  noResultsComponent={null}
/>
```

Otherwise, it feels odd that you could set `disableRenderingWithNoResults` as well as `noResultsComponent`.

Reply:
Not sure if "null" is a valid value for a property that expects a component...
Maybe we can rename it to `noResultsElement` and accept null as a value. But it might not be intuitive that when passing the null value the entire group will not be rendered instead of just keeping the icon and title but the result area empty. For this reason we added a property that explicitly states that the entire group will not render if it has no results.